tests/integration/functional/zen_stores/test_zen_store.py

```diff
@@ -0,0 +1,991 @@
+    list_of_entities,
+    ids=[e.entity_name for e in list_of_entities],
+)
+def test_updating_nonexisting_entity_raises_error(
+    crud_test_config: CrudTestConfig,
+):
+    """Tests updating a nonexistent entity raises an error."""
+    if crud_test_config.update_model:
+        # Update the created entity
+        update_model = crud_test_config.update_model
+        with pytest.raises(KeyError):
+            crud_test_config.update_method(uuid.uuid4(), update_model)
+    else:
+        pytest.skip()
+
+
+@pytest.mark.parametrize(
+    "crud_test_config",
+    list_of_entities,
+    ids=[e.entity_name for e in list_of_entities],
+)
+def test_deleting_nonexistent_entity_raises_error(
+    crud_test_config: CrudTestConfig,
+):
+    """Tests deleting a nonexistent workspace raises an error."""
+    with pytest.raises(KeyError):
+        crud_test_config.delete_method(uuid.uuid4())
+
+
+# .----------.
+# | WORKSPACES |
+# '----------'
+
+
+def test_only_one_default_workspace_present():
+    """Tests that one and only one default workspace is present."""
+    client = Client()
+    assert (
+        len(
+            client.zen_store.list_workspaces(
+                WorkspaceFilterModel(name="default")
+            )
+        )
+        == 1
+    )
+
+
+def test_updating_default_workspace_fails():
+    """Tests updating the default workspace."""
+    client = Client()
+
+    default_workspace = client.zen_store.get_workspace(DEFAULT_WORKSPACE_NAME)
+    assert default_workspace.name == DEFAULT_WORKSPACE_NAME
+    workspace_update = WorkspaceUpdateModel(
+        name="aria_workspace",
+        description="Aria has taken possession of this workspace.",
+    )
+    with pytest.raises(IllegalOperationError):
+        client.zen_store.update_workspace(
+            workspace_id=default_workspace.id,
+            workspace_update=workspace_update,
+        )
+
+
+def test_deleting_default_workspace_fails():
+    """Tests deleting the default workspace."""
+    client = Client()
+    with pytest.raises(IllegalOperationError):
+        client.zen_store.delete_workspace(DEFAULT_NAME)
+
+
+# .-------.-.
+# | TEAMS | |
+# '-------'-'
+
+
+def test_adding_user_to_team():
+    """Tests adding a user to a team."""
+    zen_store = Client().zen_store
+    team_name = "arias_team"
+    with UserContext() as created_user:
+        try:
+            new_team = TeamRequestModel(name=team_name)
+            new_team = zen_store.create_team(new_team)
+
+            team_update = TeamUpdateModel(users=[created_user.id])
+            team_update = zen_store.update_team(
+                team_id=new_team.id, team_update=team_update
+            )
+
+            assert created_user.id in team_update.user_ids
+            assert len(team_update.users) == 1
+
+            # Make sure the team name has not been inadvertently changed
+            assert zen_store.get_team(new_team.id).name == team_name
+        # Cleanup no matter what
+        finally:
```

Comment:
Isn't this what the `TeamContext`  you created is for ?

Reply:
Jup, the Conteyxts came halfway through thsi endevour, so might have missed some opportunities to backport